pkgs/by-name/ch/choose-gui/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib, stdenv, fetchFromGitHub, xcbuild, coreutils, darwin }:
+
+stdenv.mkDerivation rec {
+  pname = "choose-gui";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "chipsenkbeil";
+    repo = "choose";
+    rev = "${version}";
+    hash = "sha256-oR0GgMinKcBHaZWdE7O+mdbiLKKjkweECKbi80bjW+c=";
+  };
+
+  buildInputs = [
+    xcbuild
+    coreutils
+    darwin.apple_sdk.frameworks.Cocoa
+  ];
+
+  buildPhase = let
+    myArch = if stdenv.hostPlatform.isx86_64 then
+    "x86_64" else "arm64";
+  in ''
```

Comment:
I'd do something like:

```
let
  arch = with stdenv.hostPlatform;
    if isx86_64 then "x86_64"
    else if isAarch64 then "arm64"
    else throw "unsupported arch";
in
```

Notice, there are other Archs! What happens if one tries to use that? It's a flawed assumption to think everything else is `arm64`. Then what happens?

Reply:
This is good feedback, although, MacOS doesn't support anything outside of x86_64+ARM64 currently. So, if an alternative arch is encountered, someone's running MacOS+Nix on a (currently) impossible hardware configuration. :smile: (Makes me curious if anyone's doing this, though, now that I think about it)

Still, though, it seems like a good idea to include, so I've added this.

One question with this: since it's impossible to run modern Nix+MacOS on anything outside of 64 bit ARM64 or x86, is there any way for me to actually test `else throw "unsupported arch"`? I.E. could I somehow override `stdenv.hostPlatform`?

Reply:
No need to test that.